### PR TITLE
Upgraded symfony/console to v3.0

### PR DESCRIPTION
**Pull request by codeaid**
Upgrading Symfony console as was suggested in #144.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "mimmi20/file-loader": "~2.0",
         "symfony/filesystem": "*",
         "symfony/finder": "~2.6",
-        "symfony/console": "~2.6",
+        "symfony/console": "^3.0",
```
Comment (asgrim) [inline composer.json]: Could we use `^2.6|^3.0` to ensure nothing conflicts for now please? I can see us removing 2.x at some point in the future though.

Comment (ThaDafinser) [inline composer.json]: @asgrim if you allow both, you should also test the lowest/highest dependency.

Example:
https://github.com/ThaDafinser/UserAgentParser/blob/master/.travis.yml#L14-L15
https://travis-ci.org/ThaDafinser/UserAgentParser

Comment (asgrim) [inline composer.json]: :+1:

Comment (codeaid) [inline composer.json]: @ThaDafinser Is there something I need to change for that or is it an option in Travis that guys need to update?

Comment (asgrim) [inline composer.json]: You change it in `.travis.yml` :)

**Comment by codeaid** [comment]
I pushed an update with a modified `.travis.yml` file. I wasn't sure if I had to put in `matrix` or just `env`. Let me know it needs changing.

I also included `composer.lock` so that people using the library can run `composer install` as per the "industry standard". Even Composer developers say

> **Commit your application's `composer.lock` (along with `composer.json`) into version control.**

Source: https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file

Yet again, let me know if there's a specific reason why you wouldn't want to do it in `browscap-php`.

**Comment by ThaDafinser** [comment]
@codeaid please remove `composer.lock` . You missunderstood that line from composer devs.

The `composer.lock` needs to be added in YOUR project root. So that you install the same deps in your dev and prod enviroment.

**Comment by asgrim** [comment]
For libraries, `composer.lock` should usually *not* be committed ;)

> Note: For libraries it is not necessary to commit the lock file, see also: Libraries - Lock file.

Source: https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file

File: .travis.yml [hunk 2]
```diff
@@ -11,6 +11,11 @@ php:
...
   - linux
 
+env:
+  matrix:
+    - PREFER_LOWEST="--prefer-lowest"
+    - PREFER_LOWEST=""
+
 cache:
   directories:
   - $TRAVIS_BUILD_DIR/resources
```
Comment (mimmi20) [inline .travis.yml]: please add ` $PREFER_LOWEST` to the composer update command

**Comment by codeaid** [comment]
Should I undo the `composer.lock` addition too then?

**Comment by asgrim** [comment]
@codeaid yes please :)

**Comment by codeaid** [comment]
Done! Force-pushed to avoid unneeded comments.

File: .travis.yml [hunk 3]
```diff
@@ -22,7 +27,7 @@ before_script:
   - if [ "`phpenv version-name`" == "hhvm" ]; then echo 'opcache.enable=1' >> /etc/hhvm/php.ini; fi
   - if [ "`phpenv version-name`" == "hhvm" ]; then echo 'opcache.enable_cli=1' >> /etc/hhvm/php.ini; fi
   - travis_retry composer self-update
-  - travis_retry composer install -o --prefer-source
+  - travis_retry composer install -o --prefer-source $PREFER_LOWEST
```
Comment (asgrim) [inline .travis.yml]: This needs to change to `composer update` please (should've been that before, but this change picked up that fault!

File: .travis.yml [hunk 4]
```diff
@@ -11,6 +11,11 @@ php:
 os:
   - linux
 
+env:
+  matrix:
+    - PREFER_LOWEST="--prefer-lowest"
```
Comment (asgrim) [inline .travis.yml]: Please also use `--prefer-lowest --prefer-stable` here to avoid alpha/beta/RC usage etc.

**Comment by codeaid** [comment]
You're killing me! :) Another force push. Hopefully this time everything's fine.

**Comment by asgrim** [comment]
Thanks @codeaid, LGTM. There are issues with the lowest version of monolog being used, but I'll fix that in a separate PR :) nice work, sorry for the hassle, but thank you!

**Comment by codeaid** [comment]
No worries, James! Thanks for the the work you're doing on library!

**Pull request merged**